vault/resource_identity_group_policies.go

```diff
@@ -0,0 +1,170 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+	"github.com/terraform-providers/terraform-provider-vault/util"
+)
+
+func identityGroupPoliciesResource() *schema.Resource {
```

Comment:
Should this resource simply "ensure" that the set of policies it has is present and does not take exclusive control of all the policies for the group? (Something like [`google_project_iam_member`](https://www.terraform.io/docs/providers/google/r/google_project_iam.html#google_project_iam_member) v.s. `google_project_iam_policy`) Otherwise it does not seem to be any more useful than setting the policies directly in `vault_identity_group`.

Reply:
This separates `group_id` from group creation allowing usage of `group_id` in templated policies and then assigning them to groups. Without this resource we cannot use ids in templates.

Reply:
Sure.

It would still be useful if this resource does not take exclusive control on the list of policies.

Reply:
The use case is that I would like to have multiple `vault_identity_group_policies` in more than one place to collectively define the list of policies for the entire group.

Reply:
I am pretty sure it will lead to race conditions and unstable state, but we can try implementing single-policy management.

Reply:
I'm trying to implement single-policy management, but I am pretty sure it will lead to many hard to debug race conditions and unstable state, since policies are not external Vault object, but just a field on the Group object.

Reply:
You're right that's unfortunate.

Reply:
I've rolled back `vault_identity_group_policy` and defaulting to exclusive policy management on `vault_identity_group_policies`

Reply:
Thanks for trying this out.

The GCP API has a similar issue but they get around it by using some [mutex](https://github.com/terraform-providers/terraform-provider-google/blob/master/google/iam_project.go#L68).

Probably requires more work in the future for this.